<?xml version="1.0"?>
<launch>
    <!-- Set to use simulation time -->
    <param name="/use_sim_time" value="true"/>

    <!-- Configuration for COB, bringup simulation -->
    <include file="$(find cob_default_robot_config)/upload_param.launch" />
    <include file="$(find cob_default_env_config)/upload_param.launch" />
    <include file="$(find cob_navigation_global)/launch/2dnav_ros_dwa.launch" />
    <include file="$(find cob_bringup_sim)/launch/robot.launch" />

    <!-- Run our dynamic scene model server node -->
    <include file="$(find srs_env_model)/launch/but_envmodel_common.launch" />

    <!-- Run our GUI nodes -->
    <include file="$(find cob_interactive_teleop)/launch/interactive_teleop_sim.launch" />
    <include file="$(find srs_interaction_primitives)/launch/interaction_primitives.launch" />
    <!-- include file="$(find srs_ui_but)/launch/cob_stretch_indicator.launch" /-->
    <!-- include file="$(find srs_ui_but)/launch/distance_visualizer.launch" /-->

    <!-- Load predefined URDF model of the environment -->
    <include file="$(find srs_ui_but)/launch/$(env ROBOT_ENV)/load_urdf_model.launch" />

    <!-- PCL to depth conversion (for the projection) -->
    <node pkg="srs_ui_but" type="but_depthtoimage_node" name="but_depthtoimage_node" output="screen">
	    <remap from="pointcloud_topic_name" to="/cam3d/depth/points" />
	    <remap from="image_topic_name" to="/srs_ui_but/depth/image" />
    </node>

    <!-- Run rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-f /map -d $(find srs_ui_but)/config/projection_test.vcg" />
</launch>

